pkg/controller/controller.go

```diff
@@ -199,33 +200,34 @@ func (ctrl *CSIAttachController) pvUpdated(old, new interface{}) {
 }
 
 // syncVA deals with one key off the queue.  It returns false when it's time to quit.
-func (ctrl *CSIAttachController) syncVA() {
+func (ctrl *CSIAttachController) syncVA(ctx context.Context) {
 	key, quit := ctrl.vaQueue.Get()
 	if quit {
 		return
 	}
 	defer ctrl.vaQueue.Done(key)
 
 	vaName := key.(string)
-	klog.V(4).Infof("Started VA processing %q", vaName)
+	logger := klog.LoggerWithValues(klog.FromContext(ctx), "VolumeAttachment", vaName)
+	logger.V(4).Info("Started VolumeAttachment processing")
 
 	// get VolumeAttachment to process
 	va, err := ctrl.vaLister.Get(vaName)
 	if err != nil {
 		if apierrs.IsNotFound(err) {
 			// VolumeAttachment was deleted in the meantime, ignore.
-			klog.V(3).Infof("VA %q deleted, ignoring", vaName)
+			logger.V(3).Info("VolumeAttachment was deleted, ignoring")
 			return
 		}
-		klog.Errorf("Error getting VolumeAttachment %q: %v", vaName, err)
+		logger.Error(err, "Error getting VolumeAttachment")
 		ctrl.vaQueue.AddRateLimited(vaName)
 		return
 	}
 	if va.Spec.Attacher != ctrl.attacherName {
-		klog.V(4).Infof("Skipping VolumeAttachment %s for attacher %s", va.Name, va.Spec.Attacher)
+		logger.V(4).Info("Skipping VolumeAttachment for attacher", "attacher", va.Spec.Attacher)
 		return
 	}
-	ctrl.handler.SyncNewOrUpdatedVolumeAttachment(va)
+	ctrl.handler.SyncNewOrUpdatedVolumeAttachment(ctx, va)
```

Comment:
IMO a new context with `vaName` should be used. SyncNewOrUpdatedVolumeAttachment does not log the `vaName`, yet, this ctx does not have it (or I missed something).